### ISSUE TITLE
Chromecast can't play certain episodes - regression

### ISSUE THREAD

**Issue by agilob** [jellyfin/jellyfin#5641]
After upgrade to 10.7.x certain episodes won't start playing on chromecast. I did downgrade to 10.6.4 and the same files play correctly.

When run on 10.7.0 and 10.7.1 only first frame is loaded, video is automatically stopped and can't be resumed.

**System (please complete the following information):**
 - OS: arch, stream from android or google chrome
 - Virtualization: docker
 - Clients: chromecast
 - Jellyfin Version: 10.7.x
 - Playback: chromecast
 - Reverse Proxy: traefik
 - Base URL:  none

**Logs**
```
[17:49:07] [INF] Profile: Unknown Profile, Path: /media/the episode.mkv, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[17:49:07] [INF] Profile: Unknown Profile, No video direct play profiles found for /media/the episode.mkv with codec hevc
[17:49:07] [INF] Profile: Unknown Profile, Path: /media/the episode.mkv, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[17:49:07] [INF] Profile: Unknown Profile, No video direct play profiles found for /media/the episode.mkv with codec hevc
[17:49:07] [INF] Profile: Unknown Profile, Path: /media/the episode.mkv, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[17:49:07] [INF] Profile: Unknown Profile, No video direct play profiles found for /media/the episode.mkv with codec hevc
[17:49:08] [INF] /usr/lib/jellyfin-ffmpeg/ffmpeg -f matroska,webm -i file:"/media/the episode.mkv" -map 0:0 -map 0:1 -map -0:s -codec:v:0 libx264 -mpegts_m2ts_mode 1 -force_key_frames "expr:gte(t,n_forced*5)" -vf "scale=trunc(min(max(iw\,ih*dar)\,min(1920\,1080*dar))/2)*2:trunc(min(max(iw/dar\,ih)\,min(1920/dar\,1080))/2)*2" -copyts -avoid_negative_ts disabled -start_at_zero -pix_fmt yuv420p -preset superfast -crf 23 -maxrate 6319292 -bufsize 12638584 -profile:v:0 high -level 41 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none -vsync -1 -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 copy -y "/tmp/transcodes/transcodes/transcodes/0733af28d45be71619a6fab2dc9ee532.mkv"
[17:49:10] [WRN] Slow HTTP Response from http://192.168.1.154:8096/videos/0ba65075-c1b1-4f61-39b2-2e3ff7d7c07e/stream.mkv?DeviceId=chromecast_1616950054502&MediaSourceId=0ba65075c1b14f6139b22e3ff7d7c07e&Static=false&VideoCodec=h264,vp8&AudioCodec=ac3,aac,mp3,opus,flac,vorbis&AudioStreamIndex=1&VideoBitrate=10680000&AudioBitrate=320000&PlaySessionId=6f7907a6bb1a4fb5a50da629a3e24fad&api_key=d80471e7569a4ac293eefcd0de1ac2e6&TranscodingMaxAudioChannels=6&CopyTimestamps=true&RequireAvc=false&Tag=5318716568d4fa8db6b051fd1ccb89c2&h264-profile=high,main,baseline,constrainedbaseline&h264-level=42&aac-audiochannels=2&TranscodeReasons=VideoCodecNotSupported,AudioCodecNotSupported to 192.168.1.141 in 0:00:01.8619662 with Status Code 200
[17:49:22] [INF] Transcoding kill timer stopped for JobId d402fdbd66264ffe9f9adfc38d5b7627 PlaySessionId 6f7907a6bb1a4fb5a50da629a3e24fad. Killing transcoding
[17:49:22] [INF] Stopping ffmpeg process with q command for /tmp/transcodes/transcodes/transcodes/0733af28d45be71619a6fab2dc9ee532.mkv
[17:49:22] [INF] FFmpeg exited with code 0
[17:49:22] [INF] Deleting partial stream file(s) /tmp/transcodes/transcodes/transcodes/0733af28d45be71619a6fab2dc9ee532.mkv
```

**Comment by christovic** [comment]
Could it be this? Check the logs that appear about 20 seconds after playback fails.

https://github.com/jellyfin/jellyfin-chromecast/issues/120

**Comment by agilob** [comment]
No more logs about errors, just disconnection message:

```
[15:54:00] [INF] Playback stopped reported by app Chromecast 2.1.0 playing episode. Stopped at 0 ms
[15:54:01] [INF] Playback stopped reported by app Chromecast 2.1.0 playing episode. Stopped at 0 ms
[15:54:06] [WRN] WS 192.168.1.189 error receiving data: The remote party closed the WebSocket connection without completing the close handshake.
[15:54:06] [INF] WS 192.168.1.189 closed
[15:54:12] [INF] WS 192.168.1.189 request
```

**Comment by christovic** [comment]
Does it every produce logs that look like this?

```[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : event_playing_id     = a8ef68032e4a8969499f8b11960796c8
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : event_user_id        = 64d1fe8272fc4061ab19219607c14ef9
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : event_user_id_int    = 2
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : session_playing_id   = a8ef68032e4a8969499f8b11960796c8
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : session_user_id      = 64d1fe8272fc4061ab19219607c14ef9
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : play_method          = Transcode (v:h264 a:direct)
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : e.ClientName         = Chromecast
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : e.DeviceName         = Chromecast
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : ItemName             = This is the item name
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : ItemId               = a8ef68032e4a8969499f8b11960796c8
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : ItemType             = Episode
[13:53:17] [INF] [220] Jellyfin.Plugin.PlaybackReporting.EventMonitorEntryPoint: StartPlaybackTimer : All matches, playback registered
```

**Comment by agilob** [comment]
Nope, these two pieces of logs posted above is literally all is logged for 2 minutes of frozen playback and 2 minutes after disconnect. I can confirm the movie has AAC 5.1

**Comment by christovic** [comment]
Can you try using the 'Unstable' Chromecast version? You can find it by going to your user settings. Will have to close the tab and open it again once saved. If you're changing that setting on Android, I found that clearing the cache does the trick to get it to switch over. You'll know it's the 'Unstable' version when 'ready to cast' isn't in the middle of the screen when you connect.

**Comment by agilob** [comment]
Nope, that's even worse actually :D
 Instead screen freezing, jellyfin automatically closes connection to chromecast. Stream closes and my random background starts.

```
[16:17:18] [INF] Profile: Unknown Profile, Path: /media/episode.mkv, isEligibleForDirectPlay: False, isEligibleForDirectStream: False
[16:17:18] [INF] /usr/lib/jellyfin-ffmpeg/ffmpeg -f matroska,webm -i file:"/media/episode.mkv" -map 0:0 -map 0:1 -codec:v:0 libx264 -mpegts_m2ts_mode 1 -force_key_frames "expr:gte(t,n_forced*5)" -filter_complex "[0:3]scale=1920x1080[sub];[0:0]scale=trunc(min(max(iw\,ih*dar)\,min(1920\,1080*dar))/2)*2:trunc(min(max(iw/dar\,ih)\,min(1920/dar\,1080))/2)*2[base];[base][sub]overlay" -copyts -avoid_negative_ts disabled -start_at_zero -pix_fmt yuv420p -preset superfast -crf 23 -maxrate 6624408 -bufsize 13248816 -profile:v:0 high -level 41 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none -vsync -1 -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 copy -y "/tmp/transcodes/transcodes/transcodes/7099a5bed0127f308c36b2f974ec0391.mkv"
[16:17:19] [ERR] Error processing request: A task was canceled. URL GET /videos/6c69ca02-fa60-f9ed-06cd-51160c58b3d7/stream.mkv.
[16:17:19] [WRN] Slow HTTP Response from http://192.168.1.154:8096/videos/6c69ca02-fa60-f9ed-06cd-51160c58b3d7/stream.mkv?DeviceId=chromecast_1617030948789&MediaSourceId=6c69ca02fa60f9ed06cd51160c58b3d7&Static=false&VideoCodec=h264,vp8&AudioCodec=ac3,aac,mp3,opus,flac,vorbis&AudioStreamIndex=1&SubtitleStreamIndex=3&VideoBitrate=10680000&AudioBitrate=320000&PlaySessionId=aa3a8206660145cc9cc127a2565d01c7&api_key=d80471e7569a4ac293eefcd0de1ac2e6&SubtitleMethod=Encode&TranscodingMaxAudioChannels=6&CopyTimestamps=true&RequireAvc=false&Tag=6f2f36c8d09d9c93d7f06814d8c7b504&h264-profile=high,main,baseline,constrainedbaseline&h264-level=42&aac-audiochannels=2&TranscodeReasons=SubtitleCodecNotSupported to 192.168.1.141 in 0:00:00.7159709 with Status Code 500
[16:17:19] [INF] Stopping ffmpeg process with q command for /tmp/transcodes/transcodes/transcodes/7099a5bed0127f308c36b2f974ec0391.mkv
[16:17:19] [INF] FFmpeg exited with code 0
[16:17:19] [INF] Deleting partial stream file(s) /tmp/transcodes/transcodes/transcodes/7099a5bed0127f308c36b2f974ec0391.mkv
[16:17:20] [INF] Playback stopped reported by app Chromecast 2.1.0 playing episode. Stopped at 0 ms
[16:17:20] [WRN] Slow HTTP Response from http://192.168.1.154:8096/Sessions/Playing/Stopped to 192.168.1.141 in 0:00:01.6979405 with Status Code 204
```

After retrying to connect, mobile app crashes (feedback sent using google play thing).

**Comment by christovic** [comment]
Oh dear... Can you narrow it down to being if the file has 5.1 audio?

**Comment by cvium** [comment]
It may be related to `-mpegts_m2ts_mode 1` which erroneously enables m2ts. It's fixed by #5600, which will be included in 10.7.2.
